File: VB/UnboundColumnForCommenting/MyComments.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections

Namespace UnboundColumnForCommenting
	Public NotInheritable Class MyComments
		Private Const StorageName As String = "MyComments"

		Private Sub New()
		End Sub
		Private Shared Property Storage() As Hashtable
			Get
				Return CType(HttpContext.Current.Session(StorageName), Hashtable)
			End Get
			Set(ByVal value As Hashtable)
				HttpContext.Current.Session(StorageName) = value
			End Set
		End Property

		Public Shared Function GetComment(ByVal rowKey As Object) As String
			Dim hash As Hashtable = Storage
			If hash IsNot Nothing Then
				Return CStr(hash(rowKey))
			End If
			Return Nothing
		End Function

		Public Shared Sub SetComment(ByVal rowKey As Object, ByVal comment As String)
			Dim hash As Hashtable = Storage
			If hash Is Nothing Then
				hash = New Hashtable()
			End If
			hash(rowKey) = comment
			Storage = hash
		End Sub
	End Class
End Namespace

File: CS/UnboundColumnForCommenting/MyComments.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections;

namespace UnboundColumnForCommenting {
    public static class MyComments {
        const string StorageName = "MyComments";

        private static Hashtable Storage {
            get {
                return (Hashtable)HttpContext.Current.Session[StorageName];
            }
            set {
                HttpContext.Current.Session[StorageName] = value;
            }
        }

        public static string GetComment(object rowKey) {
            Hashtable hash = Storage;
            if(hash != null)
                return (string)hash[rowKey];
            return null;
        }

        public static void SetComment(object rowKey, string comment) {
            Hashtable hash = Storage;
            if(hash == null)
                hash = new Hashtable();
            hash[rowKey] = comment;
            Storage = hash;
        }
    }
}
